mismi-s3/test/Test/IO/Mismi/S3/Commands.hs

```diff
@@ -118,6 +118,34 @@ prop_size_recursively d = testAWS $ do
   r <- sizeRecursively (a { key = dirname $ key a })
   pure $ r === [Sized (fromIntegral . BS.length $ T.encodeUtf8 d) a]
 
+prop_join =
+  forAll arbitrary $ \bs -> (BS.length bs /= 0) ==> testAWS $ do
+    a <- newAddress
+    b <- newAddress
+    c <- newAddress
+    f <- newFilePath
+    let
+      s = f </> T.unpack "fred"
+      d = f </> T.unpack "down"
+    liftIO $ withFile s WriteMode $ \h ->
+      replicateM_ 1000 (LBS.hPut h (LBS.fromChunks . return $ (BS.concat . L.replicate 10000 $ bs)))
+    liftIO . putStrLn $ "Generated file"
+
+    uploadOrFail s a
+    uploadOrFail s b
+    liftIO . putStrLn $ "Uploaded files"
+
+    liftIO . putStrLn $ "Running copy ..."
+    eitherT (fail . show . renderConcatError) pure $ concatMultipart Fail 1 [a, b] c
+    liftIO . putStrLn $ "Done copy"
+
+    eitherT (fail . show) pure $ download c d
+    liftIO . putStrLn $ "Done download"
```

Comment:
How many times do these properties get run? These `putStrLn`s seem like a lot of output